server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -445,42 +445,38 @@ public void deleteSnapshot(SnapshotId snapshotId, long repositoryStateId, Action
-                ? listener // if we don't have any newly unreferenced indices we move to the next step directly
-                : ActionListener.wrap(
-                    vv -> deleteUnreferencedIndices(unreferencedIndices, listener), listener::onFailure);
-            deleteSnapshotBlobs(snapshot, snapshotId,
-                snapshot == null || snapshot.indices().isEmpty()
-                    ? afterDeleteIndices // if we don't have any indices to delete we move to the next step
-                    : ActionListener.wrap(
-                        v -> deleteIndices(
-                            finalSnapshotInfo.indices().stream().map(repositoryData::resolveIndexId).collect(Collectors.toList()),
-                            snapshotId, afterDeleteIndices), listener::onFailure));
-        }
-    }
-
-    private void deleteSnapshotBlobs(@Nullable SnapshotInfo snapshot, SnapshotId snapshotId, ActionListener<Void> listener) {
-        final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
-        final ActionListener<Void> deleteListener = new GroupedActionListener<>(ActionListener.map(listener, v -> null), 2);
-        executor.execute(new ActionRunnable<Void>(listener) {
-            @Override
-            protected void doRun() {
-                // delete the snapshot file
-                deleteSnapshotBlobIgnoringErrors(snapshot, snapshotId.getUUID());
-                deleteListener.onResponse(null);
-            }
-        });
-        executor.execute(new ActionRunnable<Void>(listener) {
-            @Override
-            protected void doRun() {
-                // delete the global metadata file
-                deleteGlobalMetaDataBlobIgnoringErrors(snapshot, snapshotId.getUUID());
-                deleteListener.onResponse(null);
+            try {
+                blobContainer().deleteBlobsIgnoringIfNotExists(
+                    Arrays.asList(snapshotFormat.blobName(snapshotId.getUUID()), globalMetaDataFormat.blobName(snapshotId.getUUID())));
+            } catch (IOException e) {
+                logger.warn(() -> new ParameterizedMessage("[{}] Unable to delete global metadata files", snapshotId), e);
             }
-        });
+            deleteIndices(
+                Optional.ofNullable(finalSnapshotInfo)
+                    .map(info -> info.indices().stream().map(repositoryData::resolveIndexId).collect(Collectors.toList()))
+                    .orElse(Collections.emptyList()),
+                snapshotId,
+                ActionListener.wrap(v -> {
+                    try {
+                        blobStore().blobContainer(basePath().add("indices")).deleteBlobsIgnoringIfNotExists(
+                            unreferencedIndices.stream().map(IndexId::getId).collect(Collectors.toList()));
+                    } catch (IOException e) {
+                        // a different Exception occurred while trying to delete - will just log the issue for now
```

Comment:
a different Exception? Can we update the comment.

Reply:
I just removed the comment, it's pretty pointless :)